src/Root.tsx

```diff
@@ -312,25 +317,46 @@ export default () => {
     }
   }, [appLanguage]);
 
+  // this will not be useful for errors in the key or wallet transform process
   useEffect(() => {
     const numNewKeys = Object.keys(keys).length;
     const keyLengthChange = previousKeysLength - numNewKeys;
-    if (keyLengthChange === 0) {
+    batch(() => {
```

Comment:
I think that `batch` is not needed any more for current react version: https://react-redux.js.org/api/batch